R/backport_linter.R

```diff
@@ -0,0 +1,83 @@
+#' @describeIn linters that checks for usage of unavailable functions. Not reliable for testing r-devel dependencies.
+#' @export
+backport_linter <- function(r_version = "4.0.3") {
```

Comment:
Can the default auto-detect, e.g. `getRversion()`?
Or even better for Packages, get the R version from the `Depends` field in `DESCRIPTION`? (this might be worth a separate PR, also suppressing lints for stuff imported from {backports})

Reply:
good idea, and a good direction for follow-up. Actually I think we can go beyond `backports` (there may be other backport packages) and try to check namespace imports and/or internally defined functions & skip those. New argument `skip=` defaulting to `infer_skips` where we apply some logic.

Reply:
I suggest using `except` as the argument name - it's already used in `missing_argument_linter` with the same purpose:
```
#' @param except a character vector of function names as exceptions.
```